### ISSUE TITLE
infinite recursion in bitwise operations

### ISSUE THREAD

**Issue by ntyni** [pjacklam/p5-Math-BigInt#7]
Hi,

as reported by Klaus Ethgen in https://bugs.debian.org/1012704 this hangs with recent versions of Math-BigInt:

```
% perl -Mbignum -e '1 | (1 >> 1)'
Deep recursion on subroutine "Math::BigInt::bior" at /usr/share/perl5/Math/BigFloat.pm line 3883.
Deep recursion on subroutine "Math::BigFloat::bior" at /usr/share/perl5/Math/BigInt.pm line 3513.
```

Also happens with other bitwise operations like & and ^ .

It seems to have regressed around 1.999832 (where it started to spit errors) and 1.999834 (where the errors became infinite recursion.)

**Comment by RolandRosenfeld** [comment]
I forwarded this ticket to https://rt.cpan.org/Ticket/Display.html?id=146411 since current 1.999837 is only available on CPAN (but not here at github) and BUGS file refers to rt.cpan.org.

**Comment by pjacklam** [comment]
Fixed in 1.999838, which I have now uploaded to CPAN.
